src/api/app/controllers/webui/request_controller.rb

```diff
@@ -299,6 +299,66 @@ def build_results
     @ajax_data['package'] = @action[:spkg] if @action[:spkg]
   end
 
+  def build_results_chart
+    success_data = {}
+    failed_data = {}
+    building_data = {}
+
+    # take all the build results for all the actions (only actions where it makes sense to have a build status)
+    @actions.where(type: [:submit, :maintenance_incident, :maintenance_release]).each do |action|
+      source_project_object = Project.find_by_name(action.source_project)
+      source_package_object = Package.find_by_project_and_name(source_project_object.name, action.source_package)
+
+      # fetch all build results for the source project/package
+      action_build_results_data = LocalBuildResult::ForPackage.new(
```

Comment:
@ncounter reusing this https://github.com/openSUSE/open-build-service/blob/2b928d8e8374de93addeb4d2d30da986dcdcf54c/src/api/app/models/package.rb#L856 would might save a few lines of code :)